test-result-summary-client/src/Build/PossibleIssues.jsx

```diff
@@ -6,45 +6,98 @@ import { getParams } from '../utils/query';
+    async componentDidMount() {
+        await this.fetchIssues();
+    }
+
+    async fetchIssues() {
+        const { buildName, testName } = getParams( this.props.location.search );
+
+        let additionalRepo = "";
         if (buildName.includes('j9') || buildName.includes('ibm')) {
-            issueRepos.push("eclipse/openj9");
-        }
-        issueRepos.push("AdoptOpenJDK/openjdk-tests");
-        issueRepos.push("AdoptOpenJDK/openjdk-build");
-        issueRepos.push("AdoptOpenJDK/openjdk-infrastructure");
-
-        let issueUrls = [];
-        for (let repo of issueRepos) {
-            issueUrls.push(`https://github.com/${repo}/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+${testName}`);
-        };
-        
-        const columns = [{
-            title: 'Possible Issues',
-            dataIndex: 'issues',
-            key: 'issues',
-        }];
-
-        let dataSource = [];
-        for (let index = 0; index < issueRepos.length; index++) {
-            dataSource.push({
-                key: index,
-                issues: <a href={issueUrls[index]} target="_blank" rel="noopener noreferrer">{issueRepos[index]}</a> 
-            }); 
+            additionalRepo = "+repo:eclipse/openj9";
         }
 
-        return <div>
-            <TestBreadcrumb buildId={buildId} testId={testId} testName={testName} />
-            <Table
-                columns={columns}
-                dataSource={dataSource}
-                bordered
-                title={() => testName}
-            />
-        </div>
+        fetch(`https://api.github.com/search/issues?q=${testName}+state:open+repo:AdoptOpenJDK/openjdk-tests` +
+               `+repo:AdoptOpenJDK/openjdk-infrastructure+repo:AdoptOpenJDK/openjdk-build${additionalRepo}`, {
+            method: 'get',
+        })
+        .then(response => {
+            if (response.status === 200) {
+                return Promise.resolve(response.json());
+            }
+            // If http error, e.g. 403 API Rate Limit, get detailed error message.
+            return Promise.resolve(response.json()).then((responseInJson) => {
+                return Promise.reject(responseInJson.message);
+            });
+        })
+        .then(
+            (result) => {
+                let issueTitles = [];
+                let issueUrls = [];
+                for (let issue of result.items) {
+                    issueTitles.push(issue.title);
+                    issueUrls.push(issue.html_url);
+                };
+                this.setState({
+                    isLoaded: true,
+                    issueTitles,
+                    issueUrls
+                });
+            },
+            // Http error message will be in error
+            (error) => {
+                this.setState({
+                    error
+                });
+            }    
+        )
+        .catch(error => {
+            this.setState({
+                error
+            });
+        })
+    }
```

Comment:
Please use async-await. It makes code more maintainable and readable. If you need help, please let me know.

Reply:
The new commit has been modified to use async-await, could you check it? Thanks.